proposed/2023/release-staging-and-deprecation.md

```diff
@@ -0,0 +1,199 @@
+
+Also, to avoid abuse, only a limited number of packages can be staged within a single group. While this can potentially be defined as a configurable value per organization / owner, there is no such requirement at the moment. Hence, this can be set to 1000 packages for now or not defined at all. Will leave this for potentially future consideration.
+
+When staged, packages will not be visible / accessible to anyone but the owner (including coowners). They will be able to see the current staged list for a given group using `nuget stage --list --group-id [group-id]` command. 
+
+
+#### Group Ownership
+There are few interesting scenarios related to package ownership:
+1. The user has staged a package to a group for later publishing, but the ownership has changed, and the user is no longer
+ an owner for a package in a group. At this point, publishing request should fail with a message that the group has been modified
+ from its original set, and the outcome of the request may be different from what the user was expecting, indicating the package
+ that was moved out of the group. If the user is aware of the problem and still wants to move forward with the change,
+ they will have to pass an additional `--force` parameter to the publish command: `nuget publish <group-id> --force`.
+2. The group to be published no longer has any packages in it. In this case the command should fail detailing the reason for the failure.
+3. This will be the same but for deprecation. Essentially, any bulk operation using groups, where the set of the packages in the group
+ have changed after staging, should result in an error with details about the change. The `--force` flag should again be used
+ to ignore the change and move forward with execution of the requst (either publishing or deprecation). To summarize,
+ the fundamental principle is that a user cannot deprecate a package they don't own.
+
+#### Lifespan of a staged group
+If users keep staging packages but do nothing with them, NuGet will become an ever-growing package graveyard,
+ incurring higher and higher costs over time simply for storing these packages. To avoid this problem, staging should be time-limited.
+ That is, there should be a maximum lifespan for a staged package, that is not published. NuGet should then unstage / remove
+ any package that was staged but never published within the specified / pre-configured time period. As a maximum timeslot
+ a default one month period can be considered. However, to reduce the cost of maintaining these packages,
+ the `nuget stage` command can also accept an optional `--lifespan [number of days]` parameter,
+ which can indicate a shorter time period, thus giving NuGet ability to remove those packages earlier than the maximum period.
+
+Note, that this opens up a new scenario, where a group has been created with a package 29 days ago, and then a new package is staged to the same group.
```

Comment:
This essentially mean the TTL lives on the stage package not the group, if I understand correctly. 

I think we can make this simpler and just keep the TTL on the stage itself. When any package gets pushed, the TTL gets reset. An owner could keep the package alive forever by removing then adding the same package every day. That's fine. They can come to the same effect with by deleting it entirely and creating it every day with CLI commands. It's all scriptable since we will be having CLI not UI gestures.

In light of that, I think a simpler implementation is fine.

Reply:
Agreed. I've simplified this section based on your suggestion. Thanks!